dali/operators/reader/loader/video/frames_decoder.h

```diff
@@ -117,7 +117,7 @@ class DLL_PUBLIC FramesDecoder {
    * `memory_file_size` arguments cover the entire video file, including the header.
    */
   FramesDecoder(const char *memory_file, int memory_file_size, bool build_index = true,
-                bool init_codecs = true, int num_frames = -1);
+                bool init_codecs = true, int num_frames = -1, const std::string &source_info = {});
```

Comment:
```suggestion
                bool init_codecs = true, int num_frames = -1, std::string_view source_info = {});
```
or
```suggestion
                bool init_codecs = true, int num_frames = -1, std::string source_info = {});
```
in the latter case, move source_info to filename_

Reply:
Pass it by value as suggested earlier.